### DISCUSSION TITLE
JIT Transform

### DISCUSSION THREAD

**Discussion by Drillibit** [tailwindlabs/tailwindcss#5777] [Help]
```translate-y-6``` for some reason it doesn't work, when jit is enabled.

**Comment by NovusOpus** [comment]
Can you share your tailwind.config, also a minimal example of your code? 

Because it's working fine in this [Tailwind Play](https://play.tailwindcss.com/UqYPwnJF3w)( **Jit Enabled** ).

**Reply by Drillibit** [reply]
```const path = require('path');

module.exports = {
  plugins: {
    'postcss-mixins': {},
    'postcss-nested': {},
    'postcss-import': {},
    tailwindcss: {
      config: path.resolve(__dirname, './tailwind.config.js'),
    },
    autoprefixer: {},
  },
};
```
```const mergeWith = require('lodash/mergeWith');
const isArray = require('lodash/isArray');

const theme = require('./packages/theme/src/config');
const localTheme = require('./configs/theme');

function customizer(objValue, srcValue) {
  if (isArray(objValue)) {
    return objValue.concat(srcValue);
  }
}

module.exports = {
  mode: 'jit',
  purge: ['./packages/**/*.{js,jsx,ts,tsx,css}'],
  darkMode: false,
  ...mergeWith(theme, localTheme, customizer),
};
```

**Reply by Drillibit** [reply]
.translate-y-6 {
    --tw-translate-y: 6rem;
    transform: var(--tw-transform);
}

 custom variable itself is undefined

**Reply by frederikhors** [reply]
Did you find a solution?

**Reply by pixelastronauts** [reply]
Same issue right here. Seems to have something to do with using the @apply in a CSS file and JIT not recognizing that transform is in use and setting this standard variables:
```
.translate-y-5, .transform {
    --tw-translate-x: 0;
    --tw-translate-y: 0;
    --tw-rotate: 0;
    --tw-skew-x: 0;
    --tw-skew-y: 0;
    --tw-scale-x: 1;
    --tw-scale-y: 1;
    --tw-transform: translateX(var(--tw-translate-x)) translateY(var(--tw-translate-y)) rotate(var(--tw-rotate)) skewX(var(--tw-skew-x)) skewY(var(--tw-skew-y)) scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y));
}
```

**Reply by frederikhors** [reply]
Yeah. @adamwathan suggested to try the v3 version and it works.

**Reply by pixelastronauts** [reply]
We're on v3.0.0-alpha.2

**Reply by frederikhors** [reply]
It works for me. Are you using it with PurgeCSS?

**Reply by pixelastronauts** [reply]
Found the problem, while upgrading to V3 the purge option was renamed to 'files' and all the paths were removed. Seems to be working fine now. Upgrading to V3 fixed it. 👍

**Comment by nor3nor3nor3** [comment]
i have same problem. @Drillibit Do you find a solution?
